source/core/shape.cc

```diff
@@ -1637,17 +1680,21 @@ RBFShape<dim>::initialize_bounding_box()
   Point<dim>     high_bounding_point{};
   Point<dim>     low_bounding_point{};
   Point<dim>     bounding_box_center{};
+  Point<dim>     temp_point{};
   Tensor<1, dim> half_lengths = Tensor<1, dim>();
   for (int d = 0; d < dim; d++)
     {
       high_bounding_point[d] = std::numeric_limits<double>::lowest();
       low_bounding_point[d]  = std::numeric_limits<double>::max();
       for (size_t i = 0; i < number_of_nodes; i++)
         {
+          temp_point[0]       = nodes_positions_x[i];
+          temp_point[1]       = nodes_positions_y[i];
+          temp_point[dim - 1] = nodes_positions_z[i];
```

Comment:
Is it always 3D, otherwise you are overwriting the y position using ``dim - 1``?
You could do a separate ``if constexpr (dim == 3)`` with a ``for`` loop inside (to minimize the number of ``if``).